### ISSUE TITLE
twistd web "-n" option documentation is backwards

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4176]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4176|
|Type|defect|
|Created|2009-12-17 19:52:25Z|

`twistd web --help` produces, amongst other things, this output:

```
  -n, --notracebacks      Display tracebacks in broken web pages. Displaying
                          tracebacks to users may be security risk!
```

Okay... the option is called _notracebacks_ and the behavior of supplying it is for tracebacks to be displayed?  Not likely.  Displaying tracebacks is the default, and this option really does disable that.

Attachments:

* [notracebacks.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/eda/eda51eb3dfcfb6d0135907013221a2effc029977/df7569271debf3e9cac6c677076765e469341da7.patch) (3108 bytes) - added by jesstess on 2009-12-19 20:45:09Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4176 4176
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation__easy documentation, easy
time__1261079545000000 1261079545000000
changetime__1261553261000000 1261553261000000
version__None None
owner__ 
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_1"></a>|@jesstess commented|
|-|-|

Grepping around I see:

 * twisted/python/zsh/_mktap:'(--notracebacks)-n[Display tracebacks in broken web pages. Displaying tracebacks to users may be security risk!]' \
 * twisted/python/zsh/_mktap:'(-n)--notracebacks[Display tracebacks in broken web pages. Displaying tracebacks to users may be security risk!]' \
 * twisted/python/zsh/_twistd:'(--notracebacks)-n[Display tracebacks in broken web pages. Displaying tracebacks to users may be security risk!]' \
 * twisted/python/zsh/_twistd:'(-n)--notracebacks[Display tracebacks in broken web pages. Displaying tracebacks to users may be security risk!]' \
 * twisted/web/tap.py:                ["notracebacks", "n", "Display tracebacks in broken web pages. " +

So these all probably need changing.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_3"></a>|@jesstess removed owner|
|-|-|

Display tracebacks ==> Do not display tracebacks, for the 5 cases above.

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|TimAllen commented|
|-|-|

I've reviewed this and I approve of it being committed, but I can't actually commit it myself. I'll leave this ticket in the review queue, waiting for somebody with commit access.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_6"></a>|@jesstess set status to `closed`|
|-|-|

(In [27854]) Apply notracebacks.patch

Author: jesstess
Reviewer: TimAllen
Fixes: [#4176](https://github.com/twisted/twisted/issues/4176)

twistd web '-n' documentation should say that it _disables_ tracebacks.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
